src/DIRAC/WorkloadManagementSystem/Agent/PilotLoggingAgent.py

```diff
@@ -0,0 +1,148 @@
+""" :mod: PilotLoggingAgent
+
+    PilotLoggingAgent sends Pilot log files to an SE
+"""
+
+# # imports
+import os
+from DIRAC import S_OK, S_ERROR, gConfig
+from DIRAC.ConfigurationSystem.Client.Helpers.Operations import Operations
+from DIRAC.Core.Base.AgentModule import AgentModule
+from DIRAC.DataManagementSystem.Client.DataManager import DataManager
+from DIRAC.WorkloadManagementSystem.Client.TornadoPilotLoggingClient import TornadoPilotLoggingClient
+from DIRAC.ConfigurationSystem.Client.Helpers.Registry import getVOs
+from DIRAC.Core.Utilities.Proxy import executeWithUserProxy
+
+
+class PilotLoggingAgent(AgentModule):
+    """
+    .. class:: PilotLoggingAgent
+
+    The agent sends completed pilot log files to permanent storage for analysis.
+    """
+
+    def initialize(self):
+        """
+        agent's initalisation. Use this agent's CS information to:
+        Determine what Defaults/Shifter shifter proxy to use.,
+        get the target SE name from the CS.
+        Obtain log file location from Tornado.
+
+        :param self: self reference
+        """
+
+        # configured VOs and setup
+        res = getVOs()
+        if res["OK"]:
+            self.voList = res.get("Value", [])
+        else:
+            return S_ERROR(res["Message"])
+
+        if isinstance(self.voList, str):
+            self.voList = [self.voList]
+
+        self.setup = gConfig.getValue("/DIRAC/Setup", None)
+        if self.setup is None:
+            self.log.error("Setup is not defined in the configuration")
+            return S_ERROR("Setup is not defined in the configuration")
+
+        return S_OK()
+
+    def execute(self):
+        """
+        Execute one agent cycle. Upload log files to the SE and register them in the DFC.
+        Use a shifter proxy dynamically loaded for every VO
+
+        :param self: self reference
+        """
+        voRes = {}
+        for vo in self.voList:
+            self.opsHelper = Operations(vo=vo, setup=self.setup)
+            # is remote pilot logging enabled for the VO ?
+            pilotLogging = self.opsHelper.getValue("/Pilot/RemoteLogging", True)
```

Comment:
There is an idea behind this logic. To disable remote logging for a VO is requires the option with value False, otherwise the VO will have remote logging enabled _**provided the URL is given somewhere, i.e. in Defaults.**_ There is no risk that the logging will be enabled by default in a typical installation. In other words the URL is an on/off switch, the flag is for fine tuning.